HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+
+![Token Cancel Airdrop Transaction](../assets/hip-904/frictionless-airdrop-tokenCancel.png)
+
+
+### Zero-Balance Accounts
+
+The ability for account with 0 cryptocurrency value to receive token transfers is an import DeFi experience.
+Post this HIP zero balance accounts could recieve tokens in the following ways:
+
+- A sender may send a token to an unused EVM address (alias) and the account will be auto-created without any HBAR balance. It will also be created with `max_auto_associations` set to `-1`, permitting unlimited auto-associations. Subsequent transfers of any token to this account with no hbars will succeed.
+- A wallet or other entity can create an account with `max_auto_associations` set to `-1`. Tokens can then be transferred into the account without the account owning any hbars.
+- An existing account can change `max_auto_associations` to `-1`, and then subsequent token transfers into the account do not charge the account.
+- An account with no open auto-association slots can receive airdrops but must send a `TokenClaimAirdropTransaction`, which requires a payer. If the account has zero hbars, then it can still claim the transfer if someone else is willing to pay for that transaction. For example, a Dapp could be the payer on the transaction. Both the Dapp and the account must sign the transaction.
+
+### Token Reject
+
+Today, if an account holds no balance (or no serial) of a token type, then the account can be dissociated from the token. However, tokens may have hefty custom fees for transferring a token. A user may not be able or willing to pay those fees, but cannot dissociate without doing so, effectively forcing them to pay for auto-renewal of tokens they do not want.
+
+This can be a significant security concern. A token with custom fees could  be used by a token spammer to airdrop a token to all users with a custom fee that will charge the users some HBAR or other tokens in order to get rid of the token — essentially creating a “ransom” token. To prevent this, we introduce the new `TokenRejectTransaction`.
+
+Any token (fungible or non-fungible) can be rejected, whether it was created through automatic or manual association, and no matter how long the token was held. A rejected token is transferred back to the token treasury, and all custom fees on the token are waived. The payer of the transaction will only pay the appropriate fees for the HAPI transaction type.
+
+
+![Token Reject Transaction](../assets/hip-904/frictionless-airdrop-tokenReject.png)
+
+> ⚠️ When a token is rejected, we **do not** decrement `used_auto_associations`. That field tracks the total number of times auto-associations have been made, not the current number of token types on the account that were auto-associated.
```

Comment:
Why do we not want to track the active/current number of auto associations slots Vs. the total used auto associations slots? Trying to understand this design decision. @Nana-EC

Reply:
Thanks for the question.
We do track the number of automatically associated tokens on the node.
We're just saying here that a token rejection does not affect tracking the number of tokens you've asked the network to automatically associate you with in the past.

Reply:
Thanks for the explanation!

Reply:
Np, thanks for the question :)